### PR TITLE
refactor(stream,agg): rename `AggState`

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

I'm splitting a huge change set to refactor streaming agg states, and this is the first PR.

This PR renames `AggState` to `AggStates`, cus it contains agg states for all agg calls instead of one. Later, the name `AggState` will be used to represent state for a single agg call.

## Checklist

- ~~[ ] I have written necessary rustdoc comments~~
- ~~[ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by BugenZhao** [comment]
I think only the control plane of our system uses the term `Manager`, which sounds more like a singleton and long-running stuff. 🥺 Maybe we should find a better name. Just IMO, the struct holds a set of agg states, so `AggStates` or `AggStateSet` is acceptable.

**Comment by richardchien** [comment]
> Just IMO, the struct holds a set of agg states, so `AggStates` or `AggStateSet` is acceptable.

`AggStates` sounds OK. I'm going to do something like the following in later PR, so renaming will make things more clearer.

```rust
enum AggState<S: StateStore> {
    /// State as single in-memory scalar value, e.g. `count`, `sum`, append-only `min`/`max`.
    InMemoryValueState(InMemoryValueState),

    /// State as materialized input, e.g. non-append-only `min`/`max`, `string_agg`
    MaterializedInputState(MaterializedInputState<S>),
}

pub struct AggStates<S: StateStore> {
    /// Group key of the state.
    group_key: Option<Row>,

    /// Managed agg states.
    states: Vec<AggState<S>>,

    /// Previous outputs of all managed states. Initialized with `None`.
    ///
    /// We use `Vec<Datum>` instead of `Row` here to avoid unnecessary memory
    /// usage for group key prefix and unnecessary construction of `Row` struct.
    prev_outputs: Option<Vec<Datum>>,
}
```

File: src/stream/src/executor/local_simple_agg.rs [hunk 1]
```diff
@@ -62,26 +62,30 @@ impl LocalSimpleAggExecutor {
...
                     agg_call,
                     &columns,
                     visibility.as_ref(),
                     capacity,
-                )?;
-                let cols = agg_call
+                )
+            })
+            .try_collect()?;
+        multizip((agg_calls, visibilities, states)).try_for_each(
```
Comment (BugenZhao) [inline src/stream/src/executor/local_simple_agg.rs:78]: https://docs.rs/itertools/latest/itertools/fn.multizip.html 🤔

> Prefer [izip!()](https://docs.rs/itertools/latest/itertools/macro.izip.html) over multizip for the performance benefits of using the standard library .zip(). Prefer multizip if a nameable type is needed.

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Comment by richardchien** [comment]
According to @st1page offline, `AggGroup` may be a reasonable name

**Comment by BugenZhao** [comment]
> According to @st1page offline, `AggGroup` may be a reasonable name

This sounds like the agg states in a group key. 🤔

**Comment by richardchien** [comment]
> > According to @st1page offline, `AggGroup` may be a reasonable name
> 
> This sounds like the agg states in a group key. 🤔

Yes it does, it is a struct per group by group.

File: src/stream/src/executor/global_simple_agg.rs [hunk 2]
```diff
@@ -271,15 +255,16 @@ impl<S: StateStore> GlobalSimpleAggExecutor<S> {
...
-            mut agg_state,
             agg_calls,
             extreme_cache_size,
             mut agg_state_tables,
             mut result_table,
             mut state_changed,
         } = self;
-        let mut input = input.execute();
 
+        let mut agg_group = None;
```
Comment (BugenZhao) [inline src/stream/src/executor/global_simple_agg.rs:265]: If we explicitly initialize it here before calling `apply_chunk`, maybe there's no need to pass an `Option` and `unwrap_mut` everywhere, which looks nicer. Anyway, it's not a big deal.

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: src/stream/src/executor/global_simple_agg.rs (hunk 2 above)
Comment (richardchien) [reply src/stream/src/executor/global_simple_agg.rs:265]: Current behavior is like delaying the creation of `AggGroup` to the first `apply_chunk`. If no `apply_chunk` ever called during one epoch, `AggGroup` doesn't need to be created.

**Pull request merged**